blocks/pipeline.py

```diff
@@ -25,14 +25,14 @@ def __init__(
             self.models = [models]
         self.datasource = datasource
 
-    def preload(self):
-        self.datasource.preload()
+    def load_remote(self):
+        self.datasource.load_remote()
         for model in self.models:
-            model.preload()
+            model.load_remote()
 
     def load(self) -> None:
-        for model in self.models:
-            model.load(self.id)
+        for i, model in enumerate(self.models):
+            model.load(self.id, i)
```

Comment:
I'm appending the ids in ``load()``. I couldn't find a neater way of keeping track of order, but this seems to work with minimal code!